addons/block_code/simple_nodes/simple_character/simple_character.gd

```diff
@@ -83,24 +113,46 @@ func _ready():
 
 func simple_setup():
 	add_to_group("affected_by_gravity", true)
+	_setup_actions()
 	_texture_updated()
 
 
-func get_custom_class():
-	return "SimpleCharacter"
+func _setup_actions():
+	if Engine.is_editor_hint() or InputMap.has_action("player_1_left"):
+		return
+
+	for player in PLAYER_KEYS:
+		for action in PLAYER_KEYS[player]:
+			var action_name = player + "_" + action
+			InputMap.add_action(action_name)
+
+			var device = int(player.split("_")[-1]) - 1
```

Comment:
This is OK, but it might be clearer to change `PLAYER_KEYS` to an array of dictionaries, then iterate it with something like this:

```gdscript
for i in PLAYER_KEYS.size():
    for action in PLAYER_KEYS[i]:
        var action_name = "player_%d_%s" % [i + 1, action]
        # etc.
```

You're still having to do arithmetic to adjust between 0-based array/controller indices and 1-based human-readable indices, but you wouldn't have to do this string mangling & conversion back to int.